joint_state_publisher/joint_state_publisher/joint_state_publisher

```diff
@@ -214,9 +214,14 @@ class JointStatePublisher():
                 elif name in self.dependent_joints:
                     param = self.dependent_joints[name]
                     parent = param['parent']
-                    joint = self.free_joints[parent]
                     factor = param.get('factor', 1)
                     offset = param.get('offset', 0)
+                    while parent in self.dependent_joints:
+                        param = self.dependent_joints[parent]
+                        parent = param['parent']
+                        factor *= param.get('factor', 1)
+                        # the offset is relative only to the first parent
```

Comment:
A part from the specific example, with the above forumlation the starting position of `Ja` depends not only on its offset, but also from its velocity ratio: if I set my application and later I need to improve a bit the factor of `Ja` for a faster motion (`x' = 3` in the example), I must also modify its offset to guarantee that the starting position remains unchanged (`y' = -2.4` if I want that `Ja` and `Jb` start from the same location).